File: abc/abc161/abc161_b.cs
// https://atcoder.jp/contests/abc161/submissions/11546663

using System;
using System.Linq;

namespace CSharp
{
    class Program
    {
        static void Main()
        {
            var input = Console.ReadLine().Split().Select(long.Parse).ToArray();
            var N = input[0];
            var M = input[1];

            var A = Console.ReadLine().Split().Select(long.Parse).ToArray();
            Array.Sort(A);
            Array.Reverse(A);
            var sum = A.Sum();

            for (var i = 0; i < M; i++)
            {
                if (A[i] * 4 * M < sum)
                {
                    Console.WriteLine("No");
                    return;
                }
            }
            Console.WriteLine("Yes");
        }
    }
}

File: abc/abc161/abc161_b.fs
// https://atcoder.jp/contests/abc161/submissions/11738785

let input = stdin.ReadLine().Split() |> Array.map int
let N = input.[0]
let M = input.[1]

let A = stdin.ReadLine().Split() |> Array.map int
let sum = A |> Array.sum

if ((A
     |> Array.filter (fun x -> x * 4 * M >= sum)
     |> Array.length) < M)
then "No"
else "Yes"
|> printfn "%s"
